pkg/controller/catalog/storage/mem_storage.go

```diff
@@ -16,211 +16,45 @@ limitations under the License.
 
 package storage
 
-import (
-	"fmt"
-	"log"
-	"strings"
-
-	model "github.com/kubernetes-incubator/service-catalog/model/service_controller"
-)
-
-type bindingPair struct {
-	binding    *model.ServiceBinding
-	credential *model.Credential
-}
-
 type memStorage struct {
-	brokers map[string]*model.ServiceBroker
-	// This gets fetched when a SB is created (or possibly later when refetched).
-	// It's static for now to keep compatibility, seems like this could be more dynamic.
-	catalogs map[string]*model.Catalog
-	// maps instance ID to instance
-	services map[string]*model.ServiceInstance
-	// maps binding ID to binding
-	// TODO: support looking up all bindings for a service instance.
-	bindings map[string]*bindingPair
+	brokers        *memStorageBroker
```

Comment:
Yes. An hour ago it wouldn't have, but a convo with @arschles re: #186 cleared some stuff up for me. I'm with you...

Just to clarify tho... you do not see anything in the above that is blocking for this PR, right?